cg/store/models.py

```diff
@@ -86,43 +127,45 @@ def low_priority(self):
-    is_external = Column(types.Boolean, nullable=False, default=False)
-    description = Column(types.String(256), nullable=False)
-    is_accredited = Column(types.Boolean, nullable=False)
-
-    turnaround_time = Column(types.Integer)
-    minimum_order = Column(types.Integer, default=1)
-    sequencing_depth = Column(types.Integer)
-    min_sequencing_depth = Column(types.Integer, default=0, nullable=False)
-    target_reads = Column(types.BigInteger, default=0)
-    percent_reads_guaranteed = Column(types.Integer, nullable=False)
-    sample_amount = Column(types.Integer)
-    sample_volume = Column(types.Text)
-    sample_concentration = Column(types.Text)
-    sample_concentration_minimum = Column(types.DECIMAL)
-    sample_concentration_maximum = Column(types.DECIMAL)
-    sample_concentration_minimum_cfdna = Column(types.DECIMAL)
-    sample_concentration_maximum_cfdna = Column(types.DECIMAL)
-    priority_processing = Column(types.Boolean, default=False)
-    details = Column(types.Text)
-    limitations = Column(types.Text)
-    percent_kth = Column(types.Integer, nullable=False)
-    comment = Column(types.Text)
-    is_archived = Column(types.Boolean, default=False)
-
-    created_at = Column(types.DateTime, default=dt.datetime.now)
-    updated_at = Column(types.DateTime, onupdate=dt.datetime.now)
-
-    versions = orm.relationship(
-        "ApplicationVersion", order_by="ApplicationVersion.version", back_populates="application"
+    id: Mapped[intpk]
+    tag: Mapped[str32] = mapped_column(unique=True)
+    prep_category: Mapped[PrepCategory]
+    is_external: Mapped[bool] = mapped_column(default=False)
+    description: Mapped[str256]
+    is_accredited: Mapped[bool] = mapped_column(nullable=False)
+
+    turnaround_time: Mapped[Optional[int]]
+    minimum_order: Mapped[Optional[int]] = mapped_column(default=1)
+    sequencing_depth: Mapped[Optional[int]]
+    min_sequencing_depth: Mapped[int] = mapped_column(default=0)
+    target_reads: Mapped[Optional[bigint]] = mapped_column(default=0)
+    percent_reads_guaranteed: Mapped[int]
+    sample_amount: Mapped[Optional[int]]
+    sample_volume: Mapped[Optional[text]]
+    sample_concentration: Mapped[Optional[text]]
+    sample_concentration_minimum: Mapped[Optional[dec]]
+    sample_concentration_maximum: Mapped[Optional[dec]]
+    sample_concentration_minimum_cfdna: Mapped[Optional[dec]]
+    sample_concentration_maximum_cfdna: Mapped[Optional[dec]]
+    priority_processing: Mapped[bool] = mapped_column(default=False)
+    details: Mapped[Optional[text]]
+    limitations: Mapped[Optional[text]]
+    percent_kth: Mapped[int]
+    comment: Mapped[Optional[text]]
+    is_archived: Mapped[bool] = mapped_column(default=False)
+
+    created_at: Mapped[Optional[datetime]] = mapped_column(default=datetime.now)
+    updated_at: Mapped[Optional[datetime]] = mapped_column(onupdate=datetime.now)
+
+    versions: Mapped[list["ApplicationVersion"]] = orm.relationship(
```

Comment:
Do we add these annotations for the relationships as well?

Reply:
Not sure what you mean. But this is done in their tutorial 😅